src/client/tile.cpp

```diff
@@ -2048,3 +2051,36 @@ video::SColor TextureSource::getTextureAverageColor(const std::string &name)
 	c.setAlpha(255);
 	return c;
 }
+
+video::ITexture *TextureSource::getShaderFlagsTexture(TileDef *tiledef, TileSpec *tile)
+{
+	video::ITexture *flags_texture;
+	std::string tname = "__shaderFlagsTexture";
+
+	bool normalmap_present = tile->normal_texture ? true : false;
+	tname += normalmap_present ? "1" : "0";
+	tname += tiledef->tileable_horizontal ? "1" : "0";
+	tname += tiledef->tileable_vertical ? "1" : "0";
+
+	if (isKnownSourceImage(tname)) {
+		u32 id;
+		flags_texture = getTexture(tname.c_str(), &id);
+		return flags_texture;
+	} else {
+		video::IVideoDriver *driver = m_device->getVideoDriver();
+		video::IImage *flags_image = driver->createImage(
+			video::ECF_A8R8G8B8, core::dimension2d<u32>(1, 1));
+		sanity_check(flags_image != NULL);
+		video::SColor c(
+			255,
+			normalmap_present ? 255 : 0,
+			tiledef->tileable_horizontal ? 255 : 0,
+			tiledef->tileable_vertical ? 255 : 0);
+		flags_image->setPixel(0, 0, c);
+		insertSourceImage(tname, flags_image);
+		u32 id;
+		flags_texture = getTexture(tname.c_str(), &id);
```

Comment:
You don't need .c_str() for passing getTexture since the type is const std::string &, so it'd construct a new std::string anyway.